src/develop/lightroom.c

```diff
@@ -469,560 +1009,83 @@ void dt_lightroom_import(int imgid, dt_develop_t *dev, gboolean iauto)
-    {
-      ptc_split[2] = g_ascii_strtod((char *)value, NULL) / 100.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"ToneCurveName2012"))
-    {
-      if(!xmlStrcmp(value, (const xmlChar *)"Linear"))
-        curve_kind = linear;
-      else if(!xmlStrcmp(value, (const xmlChar *)"Medium Contrast"))
-        curve_kind = medium_contrast;
-      else if(!xmlStrcmp(value, (const xmlChar *)"Strong Contrast"))
-        curve_kind = strong_contrast;
-      else if(!xmlStrcmp(value, (const xmlChar *)"Custom"))
-        curve_kind = custom;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentRed"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][0] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentOrange"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][1] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentYellow"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][2] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentGreen"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][3] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentAqua"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][4] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentBlue"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][5] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentPurple"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][6] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"SaturationAdjustmentMagenta"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[1][7] = 0.5 + (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"LuminanceAdjustmentRed"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[0][0] = 0.5 + lfactor * (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"LuminanceAdjustmentOrange"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[0][1] = 0.5 + lfactor * (float)v / 200.0;
+  // let's now parse the needed data
+
+  lr_data_t data = { 0 };
```

Comment:
Okay, yeah, except the warnings.
I think one `memset()` over the entire `struct` should be fine still.

Reply:
@LebedevRI houz said that a single "{ 0 }" was ok and on my side I have no warnings. Maybe a different compiler? I'm using GCC 7.2.

Reply:
https://travis-ci.org/darktable-org/darktable/builds/312355224?utm_source=github_status&utm_medium=notification

Reply:
Well, just to be safe we should use memset then.

Reply:
Ok, otherwise using "{ { 0 } }" works fine. But I'll use one memset().